packages/format-library/src/language/index.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+import {
+	TextControl,
+	SelectControl,
+	Button,
+	Popover,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { applyFormat, removeFormat, useAnchor } from '@wordpress/rich-text';
+import { translation } from '@wordpress/icons';
+
+const name = 'core/language';
+const title = __( 'Language' );
+
+export const language = {
+	name,
+	tagName: 'span',
+	className: 'has-language',
+	edit: Edit,
+	title,
+};
+
+function Edit( props ) {
+	const { contentRef, isActive, onChange, value } = props;
+	const popoverAnchor = useAnchor( {
+		editableContentElement: contentRef.current,
+		settings: language,
+	} );
+
+	const [ lang, setLang ] = useState( '' );
+	const [ dir, setDir ] = useState( 'ltr' );
+
+	const [ isPopoverVisible, setIsPopoverVisible ] = useState( false );
+	const togglePopover = () => {
+		setIsPopoverVisible( ( state ) => ! state );
+		setLang( '' );
+		setDir( 'ltr' );
+	};
+
+	return (
+		<>
+			<RichTextToolbarButton
+				icon={ translation }
+				label={ title }
+				title={ title }
+				onClick={ () => {
+					if ( isActive ) {
+						onChange( removeFormat( value, name ) );
+					} else {
+						togglePopover();
+					}
+				} }
+				isActive={ isActive }
+				role="menuitemcheckbox"
+			/>
+
+			{ isPopoverVisible && (
+				<Popover
+					className="block-editor-format-toolbar__language-popover"
+					anchor={ popoverAnchor }
+					placement="bottom"
```

Comment:
```suggestion
					placement="bottom"
					onClose={  togglePopover } 
```

I think we need to close the modal when focusing outside; otherwise, I see this strange behavior, when the modal isn't closed when I focus on different parts of the text.

## Screencast

https://user-images.githubusercontent.com/240569/235847827-c9d6d10e-7424-4729-9c2a-e090e553a7eb.mp4